src/graphnet/models/gnn/grit.py

```diff
@@ -0,0 +1,160 @@
+"""Implementation of GRIT, a graph transformer model.
+
+Original author: Liheng Ma
+Original code: https://github.com/LiamMa/GRIT
+Paper: "Graph Inductive Biases in Transformers without Message Passing",
+        https://arxiv.org/abs/2305.17589
+
+Adapted by: Philip Weigel
+"""
+
+import torch.nn as nn
+from torch import Tensor
+from torch_geometric.data import Data
+
+from graphnet.models.gnn.gnn import GNN
+
+from graphnet.models.components.layers import (
+    GritTransformerLayer,
+    SANGraphHead,
+)
+from graphnet.models.components.embedding import (
+    RRWPLinearEdgeEncoder,
+    RRWPLinearNodeEncoder,
+    LinearNodeEncoder,
+    LinearEdgeEncoder,
+    RWSELinearNodeEncoder,
+)
+
+
+class GRIT(GNN):
+    """GRIT is a graph transformer model.
+
+    Original code:
+    https://github.com/LiamMa/GRIT/blob/main/grit/network/grit_model.py
+    """
+
+    def __init__(
+        self,
+        nb_inputs: int,
+        hidden_dim: int,
+        nb_outputs: int = 1,
+        ksteps: int = 21,
+        n_layers: int = 10,
+        n_heads: int = 8,
+        pad_to_full_graph: bool = True,
+        add_node_attr_as_self_loop: bool = False,
+        dropout: float = 0.0,
+        fill_value: float = 0.0,
+        norm: nn.Module = nn.BatchNorm1d,
```

Comment:
Would be informative to mention here that the normalization layer is assumed to be un-instantiated. E.g: 

`norm: Uninstantiated normalization layer. Must be either BatchNorm1d or BatchNorm1d`